src/shogun/features/DenseFeatures.cpp

```diff
@@ -352,6 +355,12 @@ template<class ST> void CDenseFeatures<ST>::copy_feature_matrix(SGMatrix<ST> src
 	initialize_cache();
 }
 
+template<class ST> Eigen::Map < Eigen::Matrix< ST, -1, -1, 0,-1,-1>, 0, Eigen::Stride<0,0>> CDenseFeatures<ST>::get_eigen_map()
```

Comment:
Why should we return a reference?  If I think about it, it would be a good idea if we're dealing with a lot of features - the object could be allocated on the heap/free store and a reference could be passed back to LARs.  This would avoid having the whole thing made on the stack (as it currently is), just to have it copied on returning it.